### ISSUE TITLE
Inconsistency when parsing an InternetAddressList with `Parse` and `TryParse`

### ISSUE THREAD

**Issue by 0xced** [jstedfast/MimeKit#729]
**Describe the bug**
When parsing text into an `InternetAddressList` object, the `Parse` and `TryParse` methods may behave differently. I.e. `TryParse` can succeed and `Parse` can fail for the same input.

**Platform**
 - OS: Windows (but not relevant for this bug)
 - .NET Runtime: CoreCLR (but not relevant for this bug)
 - .NET Framework: .NET Core 3.1, .NET 5 and .NET 6
 - MimeKit Version: 2.15.1 and 3.0.0-preview1

**To Reproduce**
Run this code snippet:

```c#
const string text = "example1@mimekit.net;example2@mimekit.net";

if (InternetAddressList.TryParse(text, out var result1))
{
    Console.WriteLine($"TryParse: {result1}");
}

var result2 = InternetAddressList.Parse(text);
Console.WriteLine($"Parse: {result2}");
```

**Expected behavior**
Either both `Parse` and `TryParse` fail or both `Parse` and `TryParse` succeed to parse the same input (`example1@mimekit.net;example2@mimekit.net`)

**Actual behavior**
The `TryParse` method _succeeds_ and parses `example1@mimekit.net` but the `Parse` method fails with `MimeKit.ParseException: Invalid local-part at offset 20`

**Comment by jstedfast** [comment]
This is known and working as designed even though I do agree that it feels inconsistent with standard .NET API expectations.

**Comment by jstedfast** [comment]
The problem is that an all-or-nothing TryParse API is useless when parsing non-standards-compliant email messages.

The MimeParser uses the TryParse method(s) and needs to be able to extract as much as it can from what it is given.

**Comment by jstedfast** [comment]
I'm going to close this for the above stated reason. That said, I'm open to reconsideration.

**Comment by 0xced** [comment]
For the record, this issue was fixed by commit 52891c179ff1273ebbeb1cf0b1666c18668743a4 (fixing #762 which is very closely related) and released in [MimeKit 3.2.0](https://github.com/jstedfast/MimeKit/blob/fdff9b95fa036a3e6e2d556c1548afaaf39a27ef/ReleaseNotes.md#mimekit-320-2022-03-26).
